pkg/e2e/infra/utils.go

```diff
@@ -118,6 +126,79 @@ func getMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList
 	return &machineSetList, nil
 }
 
+// getMachineSetList returns a MachineSetList object filtered by machineRoleLabel
+func getWorkerMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList, error) {
+	machineSetList := mapiv1beta1.MachineSetList{}
+	listOptions := runtimeclient.ListOptions{
+		Namespace: e2e.TestContext.MachineApiNamespace,
+	}
+
+	// See https://github.com/openshift/cluster-api-provider-kubemark/blob/master/examples/worker-machinesets.yaml#L45
+	// We plan to update the installer and consumers to rely only on applying 'node-role.kubernetes.io/worker'to machineSets
+	// in favour of 'machine.openshift.io/cluster-api-machine-role'
+	listOptions.MatchingLabels(map[string]string{nodeWorkerRoleLabel: ""})
```

Comment:
The worker node label is part of a machine spec, not machineset labels. This will not find any machineset.

Reply:
just getting all machineSets now